Anlab.Mvc/Services/IOrderService.cs

```diff
@@ -16,22 +17,72 @@ public interface IOrderService
         Task PopulateOrder(OrderSaveModel model, Order orderToUpdate);
         void PopulateOrderWithLabDetails(OrderSaveModel model, Order orderToUpdate);
         Task SendOrderToAnlab(Order order);
+
+        List<TestItemModel> PopulateTestItemModel();
     }
 
     public class OrderService : IOrderService
     {
         private readonly ApplicationDbContext _context;
+        private readonly ITestItemPriceService _itemPriceService;
+        private readonly AppSettings _appSettings;
 
-        public OrderService(ApplicationDbContext context)
+        public OrderService(ApplicationDbContext context, ITestItemPriceService itemPriceService, IOptions<AppSettings> appSettings)
         {
             _context = context;
+            _itemPriceService = itemPriceService;
+            _appSettings = appSettings.Value;
+        }
+
+        public List<TestItemModel> PopulateTestItemModel()
+        {
+            var prices = _itemPriceService.GetPrices();
+            var items = _context.TestItems.AsNoTracking().ToList();
+
+            var joined = (from i in items
+                join p in prices on i.Code equals p.Code
+                select new TestItemModel
+                {
+                    Analysis = i.Analysis,
+                    Category = i.Category,
+                    Code = i.Code,
+                    ExternalCost = Math.Ceiling(p.Cost * _appSettings.NonUcRate),
+                    Group = i.Group,
+                    Id = i.Id,
+                    InternalCost = Math.Ceiling(p.Cost),
+                    ExternalSetupCost = Math.Ceiling(p.SetupCost * _appSettings.NonUcRate),
+                    InternalSetupCost = Math.Ceiling(p.SetupCost)
+                }).ToList();
+            return joined;
+        }
+
+        private IList<TestItemModel> PopulateSelectedTestsItemModel(IEnumerable<int> selectedTestIds)
+        {
+            var prices = _itemPriceService.GetPrices();
+            var items = _context.TestItems.Where(a => selectedTestIds.Contains(a.Id)).AsNoTracking().ToList();
+
+            var joined = (from i in items
+                join p in prices on i.Code equals p.Code
+                select new TestItemModel
+                {
+                    Analysis = i.Analysis,
+                    Category = i.Category,
+                    Code = i.Code,
+                    ExternalCost = Math.Ceiling(p.Cost * _appSettings.NonUcRate),
+                    Group = i.Group,
+                    Id = i.Id,
+                    InternalCost = Math.Ceiling(p.Cost),
+                    ExternalSetupCost = Math.Ceiling(p.SetupCost * _appSettings.NonUcRate),
+                    InternalSetupCost = Math.Ceiling(p.SetupCost)
+                }).ToList();

```

Comment:
@jSylvestre this exact code is duplicated, remember DRY.  I think in this case a good way to go would be to have TestItemModel have a ctor() which takes in an item and a price and constructs this object.  Otherwise we'll have 2 places doing the same pricing calculations, which is asking for trouble